Modules/_sqlite/module.c

```diff
@@ -77,11 +77,11 @@ pysqlite_connect_impl(PyObject *module, PyObject *database, double timeout,
     else {
         Py_INCREF(isolation_level);
     }
-    PyObject *res = PyObject_CallFunction(factory, "OdiOiOii", database,
+    PyObject *res = PyObject_CallFunction(factory, "NdiOiOii",
+                                          PyOS_FSPath(database),
```

Comment:
As you mention in https://github.com/python/cpython/pull/92926#issuecomment-1131475986, if `PyOS_FSPath` is not needed, we can revert this particular change.